lib/spack/spack/cmd/find.py

```diff
@@ -21,23 +22,18 @@
 
 def setup_parser(subparser):
     format_group = subparser.add_mutually_exclusive_group()
-    format_group.add_argument('-s', '--short',
-                              action='store_const',
-                              dest='mode',
-                              const='short',
-                              default='short',
-                              help='show only specs (default)')
-    format_group.add_argument('-p', '--paths',
-                              action='store_const',
-                              dest='mode',
-                              const='paths',
-                              help='show paths to package install directories')
     format_group.add_argument(
-        '-d', '--deps',
-        action='store_const',
-        dest='mode',
-        const='deps',
-        help='show full dependency DAG of installed packages')
+        "--format", action="store", default=None,
+        help="output specs with the specified format string")
```

Comment:
Unclear which type of format string this accepts. Could be `%f`, or `{0}`, or some color code thing. Should make it more clear what kind of string is accepted (maybe the docs are enough?)

Reply:
I think the docs are enough -- though this makes me think it would be cool to make `spack docs spack.spec.Spec.format` pop up the API docs (as we could then refer to it) .  *That* is another PR though.